cmake/patches/protobuf.patch

```diff
@@ -0,0 +1,27 @@
+diff --git a/src/google/protobuf/compiler/cpp/cpp_file.cc b/src/google/protobuf/compiler/cpp/cpp_file.cc
```

Comment:
Um... I would refrain from patching a well known library; we should probably upstream this to google/protobuf.

Reply:
Actually, I would recommend adding /wd4800 to the compiler flag, instead of explicitly patching. Also, you will encounter way more than C4800 alone, see

https://github.com/google/protobuf/tree/master/cmake

(see "Notes on Compiler Warnings")

Reply:
OK. changed to disable the warning in the project level.